evcxr/src/errors.rs

```diff
@@ -52,18 +52,41 @@ fn spans_in_local_source(span: &JsonValue) -> Option<&JsonValue> {
 fn get_code_origins_for_span<'a>(
     span: &JsonValue,
     code_block: &'a CodeBlock,
-) -> Vec<(&'a CodeKind, usize)> {
-    let mut code_origins = Vec::new();
+) -> (Vec<(&'a CodeKind, usize)>, (usize, usize)) {
     if let Some(span) = spans_in_local_source(span) {
+        let mut code_origins = Vec::new();
+
         if let (Some(line_start), Some(line_end)) =
             (span["line_start"].as_usize(), span["line_end"].as_usize())
         {
             for line in line_start..=line_end {
                 code_origins.push(code_block.origin_for_line(line));
             }
         }
+        let mut bs = span["byte_start"].as_usize().unwrap_or(0) + 20;
+        let mut be = span["byte_end"].as_usize().unwrap_or(0) + 20;
+        for x in &code_block.segments {
+            if x.code.len() > bs {
+                break;
+            }
+            if matches!(x.kind, CodeKind::OriginalUserCode(_)) {
+                break;
+            }
+            bs -= x.code.len();
+        }
+        for x in &code_block.segments {
+            if x.code.len() > be {
+                break;
+            }
+            if matches!(x.kind, CodeKind::OriginalUserCode(_)) {
+                break;
+            }
+            be -= x.code.len();
+        }
+        (code_origins, (bs, be))
```

Comment:
This code tries to covert the byte offset returned by cargo, to the byte offset inside user code. I currently tried to decrease size of code segments other than `CodeKind::OriginalUserCode` that may appear before user's code. What should I do here?

Reply:
It seems it works on single statement blocks. If I remove `+20`, the query `2 + 3.` would work correctly:
![image](https://user-images.githubusercontent.com/45197576/185761580-12f7a9ce-18fe-4e52-8c15-baf6c9be4d97.png)
But not on multiple statements blocks:
![image](https://user-images.githubusercontent.com/45197576/185761629-680ac49e-75b1-42c7-92dc-e15f11d79c32.png)

Reply:
Nevermind I think I fixed it.